### PR TITLE
P2591R5 Concatenation of strings and string views

### LINKED ISSUE #6882: [2024-03 LWG Motion 11] Concatenation of strings and string views
https://wiki.edg.com/pub/Wg21tokyo2024/StrawPolls/P2591R5.html

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #6882 
Fixes cplusplus/papers#1249

### CODE REVIEW

File: source/strings.tex [hunk 1]
```diff
@@ -4879,6 +4895,85 @@
...
+    operator+(basic_string<charT, traits, Allocator>&& lhs,
+              type_identity_t<basic_string_view<charT, traits>> rhs);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Equivalent to:
+\begin{codeblock}
+lhs.append(rhs);
+return std::move(lhs);
```
Comment (frederick-vs-ja) [inline source/strings.tex:4929]: Should we drop `std::move()` here as `lhs` is move-eligible? (ditto below)

Comment (jensmaurer) [reply source/strings.tex:4929]: @jwakely , any opinion here?

Comment (tkoeppe) [reply source/strings.tex:4901]: Since this makes no difference here, let's stick with what was approved for the paper.

Comment (jwakely) [reply source/strings.tex:4901]: Yeah I don't mind either way.

**Pull request merged**